pkg/southbound/gnmi/conn_manager.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2021-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmi
+
+import (
+	"context"
+	"sync"
+
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+)
+
+// ConnManager gNMI connection manager
+type ConnManager interface {
+	Get(ctx context.Context, id ConnID) (Conn, error)
+	List(ctx context.Context) ([]Conn, error)
+	Watch(ctx context.Context, ch chan<- Conn) error
+	Add(conn Conn) error
```

Comment:
I actually think we may just want to have a `Connect` method on the `ConnManager` interface. This way the manager can ensure there's only one connection to a specific target. The way the interface is defined now, the connection is actually created before it's added to the manager, and this will allow multiple connections to be opened to the same device simultaneously. This is something we can and should prevent with good API design.